src/perf/bin/drvmain.cpp

```diff
@@ -585,9 +602,15 @@ QUIC_THREAD_CALLBACK(PerformanceWaitForStopThreadCb, Context)
         return;
     }
 
-    WdfRequestUnmarkCancelable(Request);
+    WdfObjectAcquireLock(Request);
+    Status = WdfRequestUnmarkCancelable(Request);
+    Client->CleanupHandleCancellation = TRUE;
+    WdfObjectReleaseLock(Request);
```

Comment:
Calling out while holding a lock? Generally a really bad idea...

Reply:
The driver framework is designed to do this. It's an option to enable automatic synchronization of all callbacks, but we don't need quite that deep. But we absolutely do need to synchronize between out thread and a call to cancel.